fmt/src/selection.rs

```diff
@@ -75,13 +75,14 @@ impl Selection {
 
         let (excludes, reverse_excludes) = {
             let mut excludes = self.excludes;
-            let reverse_excludes = excludes
-                .extract_if(|pat| pat.starts_with('!'))
-                .map(|mut pat| {
-                    pat.remove(0);
-                    pat
-                })
-                .collect::<Vec<_>>();
+            let mut reverse_excludes = Vec::new();
```

Comment:
I tend to reimpl drain_filter as a function and add a to-do that we can replace it once the unstable feature becomes stable.

Reply:
Can you elaborate? `drain_filter` is an unstable feature, no? Do you mean wrapping this functionality in another function?

Reply:
I found it's not easy and not worthy to add a generic function, so I just add a todo on the block - https://github.com/korandoru/hawkeye/pull/132/commits/c574d1081b939069267268b36c446199692a6442